core/modules/core/include/klepsydra/core/unsafe_multi_listener_event_emitter.h

```diff
@@ -0,0 +1,225 @@
+            std::string listenerKeyName = listenerName;
+            _listenersNameMap[listenerKeyName] = listenerId;
+            if (container) {
+                auto listenerStat = std::make_shared<SubscriptionStats>(subscriberName,
+                                                                        listenerName,
+                                                                        "EVENT_EMITTER");
+                _listenerStats[listenerId] = listenerStat;
+                container->attach(listenerStat.get());
+                auto wrappedCallback = [listenerStat,
+                                        callback](const Event &event,
+                                                  long long unsigned int enqueuedTimeNs) {
+                    listenerStat->startProcessMeasure();
+                    if (enqueuedTimeNs > 0) {
+                        listenerStat->totalEnqueuedTimeInNs +=
+                            (TimeUtils::getCurrentNanosecondsAsLlu() - enqueuedTimeNs);
+                    }
+                    callback(event);
+                    listenerStat->stopProcessMeasure();
+                };
+                _listeners.push_back(wrappedCallback);
+
+            } else {
+                auto wrappedCallback = [callback](const Event &event,
+                                                  long long unsigned int enqueuedTimeNs) {
+                    callback(event);
+                };
+                _listeners.push_back(wrappedCallback);
+            }
+            _subscriberNameMap[subscriberName].push_back(listenerId);
+            return listenerId;
+        }
+    }
+
+    /**
+     * @brief once
+     * @param subscriberName
+     * @param callback
+     * @return
+     */
+    unsigned int once(const std::string &subscriberName,
+                      std::function<void(const Event &event)> callback) override
+    {
+        throw std::invalid_argument(
+            "kpsr::UnsafeMultiListenerEventEmitter::once: Unsupported operation.");
+    }
+
+    /**
+     * @brief removeListener
+     * @param listenerId
+     */
+    void removeListener(Container *container, unsigned int listenerId) override
+    {
+        spdlog::debug("{} listenerId: {}", __PRETTY_FUNCTION__, listenerId);
+        if (_listeners.size() > listenerId) {
+            _listeners[listenerId] = nullptr;
+            std::string listenerName;
+            bool found = false;
+            for (auto &it : _listenersNameMap) {
+                if (it.second == listenerId) {
+                    listenerName = it.first;
+                    found = true;
+                    break;
+                }
+            }
+            if (found) {
+                _listenersNameMap.erase(listenerName);
+                if (container && (_listenerStats.find(listenerId) != _listenerStats.end())) {
+                    container->detach(_listenerStats[listenerId].get());
+                    _listenerStats.erase(listenerId);
+                }
```

Comment:
I think we should add the subscriberNameMap removal as in the PR https://github.com/klepsydra-technologies/kpe-core/pull/48/files

Reply:
Updated to include this PR